go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -165,59 +178,138 @@ func (oc *Controller) updateEgressFirewallWithRetry(egressfirewall *egressfirewa
 	})
 }
 
-func (ef *egressFirewall) addACLToJoinSwitch(joinSwitches []string, hashedAddressSetNameIPv4, hashedAddressSetNameIPv6 string) error {
+func (oc *Controller) addACLToJoinSwitch(joinSwitches []string, hashedAddressSetNameIPv4, hashedAddressSetNameIPv6 string, ef *egressFirewall) error {
 	for _, rule := range ef.egressRules {
-		var match string
 		var action string
+		var match string
 		if rule.access == egressfirewallapi.EgressFirewallRuleAllow {
 			action = "allow"
 		} else {
 			action = "drop"
 		}
-
-		ipAddress, _, err := net.ParseCIDR(rule.to.cidrSelector)
-		if err != nil {
-			return fmt.Errorf("error rule.to.cidrSelector %s is not a valid CIDR (%+v)", rule.to.cidrSelector, err)
-		}
-		if !utilnet.IsIPv6(ipAddress) {
-			match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv4)
+		if rule.to.cidrSelector != "" {
+			var err error
+			match, err = generateMatch(hashedAddressSetNameIPv4, hashedAddressSetNameIPv6, []string{rule.to.cidrSelector}, rule.ports)
+			if err != nil {
+				return err
+			}
 		} else {
-			match = fmt.Sprintf("match=\"ip6.dst == %s && ip6.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv6)
+			// rule based on DNS NAME
+			if rule.access == egressfirewallapi.EgressFirewallRuleDeny {
+				return fmt.Errorf("do not use dnsName as part of a deny rule this is insecure")
+			}
+			if oc.egressFirewallDNS == nil {
+				var err error
+				oc.egressFirewallDNS, err = NewEgressDNS()
+				if err != nil {
+					return err
+				}
+				// in the event that the global stopChannel is called close the egressFirewall stopChannel
+				go utilwait.Until(func() {
+					<-oc.stopChan
+					close(oc.egressFirewallDNS.stopChan)
+				}, 0, oc.stopChan)
+			}
+			dnsNameAddressSets, err := oc.addressSetFactory.NewAddressSet(rule.to.dnsName, nil)
```

Comment:
This should be managed by the egressFirewallDNS controller; there should be one address set per dnsName across the whole cluster.

Reply:
moved this code to inside the CRD watcher